.gitconfig

```diff
@@ -57,6 +57,13 @@
 	# a.k.a. ‘delete merged’
 	dm = "!git branch --merged | grep -v '\\*' | xargs -n 1 git branch -d"
 
+	# Instantly clean a git repo from extraneous files in a non-blocking fashion
+	purge = "!f() { \
+		declare tmp_dir=$(mktemp -d); \
+		git clean -fdxn | sed -n -r 's/^Would remove (.*)/\\1/p' | xargs mv -t \"$tmp_dir\"; \
```

Comment:
or, to not invoke `mv` for every file (from [Apple's documentation](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/xargs.1.html)):

```
... | xargs -J % mv % \"$tmp_dir\"; \
```